File: baekjoon/p02526/Main.java
package baekjoon.p02526;

import java.util.ArrayList;
import java.util.List;
import java.util.Scanner;

public class Main {

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        System.out.println(solve(sc.nextInt(), sc.nextInt()));
    }

    private static int solve(int n, int p) {
        List<Integer> list = new ArrayList<>();
        int x = n;

        while (true) {
            x = (x * n % p);
            int idx = list.indexOf(x);
            if (idx != -1) {
                return (list.size() - idx);
            }

            list.add(x);
        }
    }

}

File: baekjoon/p02526/Main.kt
package baekjoon.p02526

fun main() {
    val (n, p) = readLine()!!.split(" ").map { it.toInt() }
    println(solve(n, p))
}

private fun solve(n: Int, p: Int): Int {
    val list = ArrayList<Int>()
    var x = n

    while (true) {
        x = (x * n % p)
        val idx = list.indexOf(x)
        if (idx != -1) {
            return (list.size - idx)
        }
        list.add(x)
    }
}